### PR TITLE
feat: deployment health server

### LINKED ISSUE #2: feat: Deployment health server
Deployment health server at `subgraphs/health/:deployment` to 
  - [ ] query graph node status endpoint for deployment's indexing status
  - [ ] process graph node's block number results for formatted response

### PR DESCRIPTION
**Pull request by hopeyen**
resolves #2

### CODE REVIEW

File: service/src/server/routes/deployment.rs [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+                internal_server_error_response("Unknown error")
+            }
+        }
+        Err(e) => internal_server_error_response(&e.to_string()),
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct IndexingStatus {
+    health: String,
```
Comment (aasseman) [inline service/src/server/routes/deployment.rs:87]: Could it make sense to go the extra mile and use an Enum here?

```rs
#[derive(Debug, Clone, Serialize, Deserialize)]
#[allow(non_camel_case_types)] // Need exact field names to match with GQL response
enum SubgraphHealth {
    healthy,
    unhealthy,
    failed
}
```

File: service/src/server/routes/mod.rs [hunk 2]
```diff
@@ -37,3 +40,13 @@ pub fn bad_request_response(error_body: &str) -> Response {
     )
         .into_response()
 }
+
+/// Create response for an internal server error
+pub fn internal_server_error_response(error_body: &str) -> Response {
+    (
+        StatusCode::INTERNAL_SERVER_ERROR,
+        axum::response::AppendHeaders([(HeaderName::from_static("graph-attestable"), "false")]),
```
Comment (aasseman) [inline service/src/server/routes/mod.rs:48]: I don't think that header makes sense here

**Review by aasseman — CHANGES_REQUESTED** [review]

File: service/src/server/routes/deployment.rs [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                }
+
+                if let Ok((latest, head)) = block_numbers(status) {
+                    if latest > head - 5 {
+                        (
+                            StatusCode::OK,
+                            axum::response::AppendHeaders([(
+                                HeaderName::from_static("graph-attestable"),
+                                "true",
+                            )]),
```
[the remark this answers is not in the document]
Comment (hopeyen) [reply service/src/server/routes/deployment.rs:65]: ah yessss, will remove the graph-attestable headers

File: service/src/server/routes/deployment.rs (hunk 1 above)
Comment (hopeyen) [reply service/src/server/routes/deployment.rs]: another idea is maybe we could use [serde alias](https://serde.rs/field-attrs.html), but I'm fine with the current way:)

**Pull request merged**